server/athenian/api/internal/settings.py

```diff
@@ -638,6 +638,27 @@ def repo_exists(self, logical_repo: RepositoryName) -> bool:
             return False
         return logical_repo.unprefixed in physical_repo_settings.logical_repositories
 
+    def explode_user_repos(self, repos: Iterable[str]) -> list[str]:
+        """Explode the list of repositories requested by user.
+
+        Each physical repo is exploded with its logical components if none of them is in the list.
+        Exploded physical repos are left in the result.
+        """
+        names = [RepositoryName.from_unprefixed(r) for r in repos]
+        referred_physical = {n.unprefixed_physical for n in names if n.is_logical}
+
+        exploded = []
+        for name in names:
+            exploded.append(unprefixed := name.unprefixed)
+            if (not name.is_logical) and name.unprefixed not in referred_physical:
+                try:
+                    logical_repos = self.prs(name.unprefixed).logical_repositories
+                except KeyError:
```

Comment:
We shouldn't handle a `KeyError` here. If we catch it, we did a poor job validating the repositories beforehand. We should loudly crash here and make the calling code satisfy the contract.

Reply:
At least in unit tests repositories with no logical components are not found in `self._prs` so `KeyError` happens also for valid physical repositories, which should be left untouched. I can replace this with a `and name.unprefixed in self._prs` though